core/node/node_framework/src/implementations/layers/eth_sender/aggregator.rs

```diff
@@ -95,7 +102,44 @@ impl WiringLayer for EthTxAggregatorLayer {
     }
 
     async fn wire(self, input: Self::Input) -> Result<Self::Output, WiringError> {
+        tracing::info!(
+            "Wiring tx_aggregator in {:?} mode which is {}",
+            self.settlement_mode,
+            self.settlement_mode.is_gateway()
+        );
+        tracing::info!("Contracts: {:?}", self.contracts_config);
+        tracing::info!("Gateway contracts: {:?}", self.gateway_contracts_config);
         // Get resources.
+
+        let (validator_timelock_addr, multicall3_addr, diamond_proxy_addr) =
+            if self.settlement_mode.is_gateway() {
+                (
+                    self.gateway_contracts_config
+                        .clone()
+                        .unwrap()
+                        .validator_timelock_addr,
+                    self.gateway_contracts_config
+                        .clone()
+                        .unwrap()
+                        .multicall3_addr,
+                    self.gateway_contracts_config
+                        .clone()
+                        .unwrap()
+                        .diamond_proxy_addr,
+                )
+            } else {
+                (
+                    self.contracts_config.validator_timelock_addr,
+                    self.contracts_config.l1_multicall3_addr,
+                    self.contracts_config.diamond_proxy_addr,
+                )
+            };
+
+        let eth_client = if self.settlement_mode.is_gateway() {
+            input.eth_client_gateway.unwrap().0
```

Comment:
Should also have error handling; it's not that hard to implement, just use `anyhow::Context`.